language-support/java/codegen/src/it/java-1.3/com.digitalasset/MapTest.java

```diff
@@ -0,0 +1,269 @@
+        assertEquals(fromConstructor, fromRoundtrip);
+
+    }
+
+    @Test
+    public void mapRecordVariantRoundtripTest() {
+
+        ValueOuterClass.Variant protoVariant = ValueOuterClass.Variant.newBuilder()
+                .setConstructor("RecordVariant")
+                .setValue(ValueOuterClass.Value.newBuilder()
+                        .setRecord(ValueOuterClass.Record.newBuilder()
+                                .addFields(ValueOuterClass.RecordField.newBuilder()
+                                        .setLabel("x")
+                                        .setValue(ValueOuterClass.Value.newBuilder()
+                                                .setMap(ValueOuterClass.Map.newBuilder()
+                                                        .addEntries(ValueOuterClass.Map.Entry.newBuilder()
+                                                                .setKey(ValueOuterClass.Value.newBuilder().setText("key"))
+                                                                .setValue(ValueOuterClass.Value.newBuilder().setInt64(42L))
+                                                                .build())
+                                                        .build())
+                                                .build())
+                                        .build())
+                                .build())
+                        .build())
+                .build();
+
+        Variant dataVariant = Variant.fromProto(protoVariant);
+        RecordVariant<?> fromValue = RecordVariant.fromValue(dataVariant);
+        RecordVariant<?> fromConstructor = new RecordVariant<>(Collections.singletonMap("key", 42L));
+        RecordVariant<?> fromRoundtrip = RecordVariant.fromValue(fromConstructor.toValue());
+
+        assertEquals(fromValue, fromConstructor);
+        assertEquals(fromConstructor.toValue(), dataVariant);
+        assertEquals(fromConstructor, fromRoundtrip);
+    }
+
+    @Test
+    public void mapParameterizedVariantRoundtripTest() {
+
+        ValueOuterClass.Variant protoVariant = ValueOuterClass.Variant.newBuilder()
+                .setConstructor("ParameterizedVariant")
+                .setValue(ValueOuterClass.Value.newBuilder()
+                        .setMap(ValueOuterClass.Map.newBuilder()
+                                .addEntries(ValueOuterClass.Map.Entry.newBuilder()
+                                        .setKey(ValueOuterClass.Value.newBuilder().setText("key"))
+                                        .setValue(ValueOuterClass.Value.newBuilder().setInt64(42L))
+                                        .build())
+                                .build())
+                        .build())
+                .build();
+
+        Variant dataVariant = Variant.fromProto(protoVariant);
+        ParameterizedVariant<?> fromValue = ParameterizedVariant.fromValue(dataVariant,
+                f -> f.asInt64().orElseThrow(() -> new IllegalArgumentException("Expected Long value")));
```

Comment:
To make it work with `Long` we need to add a `getValue()` after `orElseThrow()`.